### DISCUSSION TITLE
LabelEncodeIntegerGraded in Multi-instance tutorial

### DISCUSSION THREAD

**Discussion by yarkinslav** [Project-MONAI/tutorials#1488] [Q&A] [answered]
Since https://github.com/Project-MONAI/tutorials/discussions/1478, I was able to implement pseudo-labels for my project. However, I found out that I Struggle to understand the concept of [LabelEncoderIntegerGraded](https://github.com/Project-MONAI/tutorials/blob/c4ff99931b6d3248c0ad429859af16de808ece9a/pathology/multiple_instance_learning/panda_mil_train_evaluate_pytorch_gpu.py#L200), why is a reason in encoding ones up to label index, instead of simple one-hot encoding?

Thanks in advance

**Comment by KumoLiu** [comment]
Hi @relyativist, 

Regarding Multiple Instance Learning (MIL), it's important to note that we lack individual labels for all the patches within the dataset. Instead, we can only assign a label to the entire bag. In my opinion, there's no necessity to apply a one-hot encoding to this label. This is because the label essentially consists of an integer value, and our objective is simply to transform this integer label into an encoded array representation with a length equal to the number of classes.

I hope it helps, thanks!

**Reply by yarkinslav** [reply, marked as answer]
@KumoLiu Thank for your reply, and apologize for a long answer. I haven't seen up-to-index label encoding, before. It still a challenge to understand why you used BCELossWithLogits? Since you are solving multi-class classification problem, the model should assign  maximum probability to one class. From you example, [criterion is applied](https://github.com/Project-MONAI/tutorials/blob/799f40bfe528a932e5e56269360f026c4eb14033/pathology/multiple_instance_learning/panda_mil_train_evaluate_pytorch_gpu.py#L157) on output of the linear layer,  sigmoid() inside the BCE loss is applied to each of the values in linear tensor, so we can have multiple values with high logits values. Seems, that this loss is tries to solve multi-label classification rather than multi-class problem.
